inject/src/main/java/io/micronaut/context/BeanDefinitionDelegate.java

```diff
@@ -199,6 +206,7 @@ public T build(BeanResolutionContext resolutionContext, BeanContext context, Bea
             for (String key : attributes.keySet()) {
                 resolutionContext.remove(key);
             }
+            resolutionContext.putAll(oldAttributes);
```

Comment:
I'm not certain about this. Unfortunately if the resolution context already contains the attributes, they are removed  in the end. This change will put the resolution context back to what it was. This change was necessary because `((DefaultBeanContext) context).getBean(resolutionContext, argumentType, qualifier);` caused the context named attribute to be removed and further configuration retrieval was failing